model_compression_toolkit/qunatizers_infrastructure/pytorch/quantize_wrapper.py

```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================f
+
+
+from model_compression_toolkit.core.common.constants import FOUND_TORCH
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.qunatizers_infrastructure.common.node_quantization_dispatcher import \
+    NodeQuantizationDispatcher
+
+if FOUND_TORCH:
+    import torch
+    import torch.nn as nn
+
+    DISPATCHER = "dispatcher"
+    LAYER = "layer"
+    STEPS = "optimizer_step"
+    TRAINING = "training"
+
+
+    def _weight_name(name: str) -> str:
+        """Extracts the weight name from the full TensorFlow variable name.
+
+        For example, returns 'kernel' for 'dense_2/kernel:0'.
+
+        Args:
+          name: TensorFlow variable name.
+
+        Returns:
+          Extracted weight name.
+        """
+        return name.split(':')[0].split('/')[-1]
+
+
+    class PytorchQuantizationWrapper(nn.Module):
+        def __init__(self,
+                     module: nn.Module,
+                     dispatcher: NodeQuantizationDispatcher):
+            """
+            Pytorch Quantization Wrapper takes a pytorch module and dispatcher and infer a quantized module.
+
+            Args:
+                layer: A pytorch module.
+                dispatcher: A node quantization dispatcher.
+            """
+            super().__init__()
+            self.dispatcher = dispatcher
+            self.add_module('layer', module)
+            self._weight_vars = []
+            for name, quantizer in self.dispatcher.weight_quantizers.items():
+                weight = getattr(self.layer, name)
+                quantizer.initialize_quantization(weight.shape,
+                                                  name+'_quant', self)
+
+                self._weight_vars.append((name, weight, quantizer))
+
+
+        def set_quantize_weights(self, quantized_weights: dict):
+            """
+            This function update layer weights after quantization.
+
+            Args:
+                quantized_weights: a dict of weight to update
+
+            Returns: None
+
+            """
+            for weight_attr in self.dispatcher.weight_quantizers.keys():
+                weight = quantized_weights.get(weight_attr)
+                current_weight = getattr(self.layer, weight_attr)
+                if current_weight.shape != weight.shape:
+                    Logger.error(
+                        f"Existing layer weight shape {current_weight.shape} is incompatible with provided weight shape {weight.shape}")  # pragma: no cover
+
+                setattr(self.layer, weight_attr, torch.nn.Parameter(weight))
+
+        def forward(self, x: torch.Tensor) -> torch.Tensor:
+            """
+            PytorchQuantizationWrapper forward functions
+            Args:
```

Comment:
Fix args (in documentation or in function signature, whichever is the correct one)